dynatrace/workflow-commit-prediction.tf

```diff
@@ -20,47 +20,114 @@ resource "dynatrace_automation_workflow" "commit_prediction" {
+               export default async function ({execution_id}) {
+                 const ex = await execution(execution_id);
+                 const event = ex.params.event;
 
-            const response = await fetch(url, {
-              method: 'GET',
-              headers: {
-                'Authorization': `Bearer $${apiToken}`
-              }
-            }).then(response => response.json());
+                 const apiToken = await credentialVaultClient.getCredentialsDetails({
+                   id: "${dynatrace_credentials.github_pat.id}",
+                 }).then((credentials) => credentials.token);
 
-            const searchResult = response.items[0];
+                 // Search for file
+                 const url = 'https://api.github.com/search/code?q=' +
+                   `"${var.annotation_prefix}/uuid:%20'$&{event['kubernetes.predictivescaling.target.uuid']}'"` +
+                   `+repo:$&{event['kubernetes.predictivescaling.target.repository']}` +
+                   `+language:YAML`
+               /*
+                 const response = await fetch(url, {
+                   method: 'GET',
+                   headers: {
+                     'Authorization': `Bearer $${apiToken}`
+                   }
+                 }).then(response => response.json());
 
-            // Get default branch
-            const repository = await fetch(searchResult.repository.url, {
-              method: 'GET',
-              headers: {
-                'Authorization': `Bearer $${apiToken}`
-              }
-            }).then(response => response.json());
+                const searchResult = response.items[0];
 
-            return {
-              owner: searchResult.repository.owner.login,
-              repository: searchResult.repository.name,
-              filePath: searchResult.path,
-              defaultBranch: repository.default_branch
-            }
-          }
+
+                 // Get default branch
+                 const repository = await fetch(searchResult.repository.url, {
+                   method: 'GET',
+                   headers: {
+                     'Authorization': `Bearer $${apiToken}`
+                   }
+                 }).then(response => response.json());
+               */
+                 const filepath= getFilePath(event['kubernetes.predictivescaling.target.uuid'])
+                 let split= event['kubernetes.predictivescaling.target.repository'].split("/")
+                 const repo = split[1];
+                 const owner = split[0];
+                 const branch= "main"
+
+                 return {
+                   owner: owner,
+                   repository: repo,
+                   filePath: filepath,
+                   defaultBranch: branch
+                 }
+               }
+
+
+               const getFilePath = (id) => {
+
+                 switch (id)
+                 {
+                     case "23df7c56-989e-4931-b8a0-b8c9c3090c47":
```

Comment:
We should not hardcode the filepaths in the demo. That breaks the purpose of many things in here.